getting_started/build-guide/build-from-scratch.rst

```diff
@@ -125,97 +129,116 @@ parameter.
 
 .. code-block:: bash
 
    ./bootstrap
-   ./configure --prefix=`pwd`
+   ./configure --prefix=$(pwd)
    make
    make install
 
-Copy the config files to .config and build the cross compiler for your target
-platforms.
+Toolchains
+----------
+
+The config files provided refer to ``../xtensa-overlay/`` and point at
+different ``./build/xtensa-*-elf`` subdirectories. Copy the ones you
+want to ``.config`` and build the cross-compiler(s) for your target
+platform(s). ``./ct-ng build`` requires an network connection to
+download gcc components.
 
 .. code-block:: bash
 
-   #Baytrail/Cherrytrail
+   # Baytrail/Cherrytrail
    cp config-byt-gcc8.1-gdb8.1 .config
    ./ct-ng build
-   #Haswell/Broadwell
+   # Haswell/Broadwell
    cp config-hsw-gcc8.1-gdb8.1 .config
    ./ct-ng build
-   #Apollo Lake
+   # Apollo Lake
    cp config-apl-gcc8.1-gdb8.1 .config
    ./ct-ng build
-   #Cannon Lake
+   # Cannon Lake
    cp config-cnl-gcc8.1-gdb8.1 .config
    ./ct-ng build
-   #i.MX8/i.MX8X
+   # i.MX8/i.MX8X
    cp config-imx-gcc8.1-gdb8.1 .config
    ./ct-ng build
-   #i.MX8M
+   # i.MX8M
    cp config-imx8m-gcc8.1-gdb8.1 .config
    ./ct-ng build
 
-Update an environment variable to refer to the alsa-lib with the one you've
-just built.
+``./ct-ng`` is a Linux kernel style Makefile; so the sample commands below
+can be used to fix some out of date ``config-*-gcc8.1-gdb8.1`` file or find
+default values missing from it:
 
 .. code-block:: bash
 
-   export LD_LIBRARY_PATH=~/work/sof/alsa-lib/src/.libs:$LD_LIBRARY_PATH
+   ./ct-ng help
+   cp config-apl-gcc8.1-gdb8.1 .config
+   ./ct-ng oldconfig V=1
+   diff -u config-apl-gcc8.1-gdb8.1 .config
 
-Copy all five cross-compiler toolchains to ~/work/sof/.
+"Install" toolchains by copying them to ``~/work/sof/``.
 
 .. code-block:: bash
 
    ls builds/
-   #xtensa-apl-elf  xtensa-byt-elf   xtensa-cnl-elf   xtensa-hsw-elf  xtensa-imx-elf  xtensa-imx8m-elf
+   # xtensa-apl-elf  xtensa-byt-elf   xtensa-cnl-elf   xtensa-hsw-elf  xtensa-imx-elf  xtensa-imx8m-elf
    cp -r builds/* ~/work/sof/
 
 .. note::
 
-   |HSW| and |BDW| share the same cross compiler toolchain: xtensa-hsw-elf
+   |HSW| and |BDW| share the same toolchain: xtensa-hsw-elf
 
-   |BYT| and |CHT| share the same cross compiler toolchain: xtensa-byt-elf
+   |BYT| and |CHT| share the same toolchain: xtensa-byt-elf
 
-   i.MX8 and i.MX8X share the same cross compiler toolchain: xtensa-imx-elf
+   i.MX8 and i.MX8X share the same toolchain: xtensa-imx-elf
 
-Add these compilers to your PATH variable.
+Add your toolchains to your PATH variable.
 
 .. code-block:: bash
 
-   export PATH=~/work/sof/xtensa-byt-elf/bin/:$PATH
-   export PATH=~/work/sof/xtensa-hsw-elf/bin/:$PATH
-   export PATH=~/work/sof/xtensa-apl-elf/bin/:$PATH
-   export PATH=~/work/sof/xtensa-cnl-elf/bin/:$PATH
-   export PATH=~/work/sof/xtensa-imx-elf/bin/:$PATH
-   export PATH=~/work/sof/xtensa-imx8m-elf/bin/:$PATH
+   PATH=~/work/sof/xtensa-byt-elf/bin/:$PATH
```

Comment:
are you sure the export is not needed?
export will also work with in scripts and bashrc

Reply:
PATH is always exported, otherwise nothing would work. See commit message.
```
declare -p PATH
```

Reply:
See `./scripts/xtensa-build-all.sh`, it extends PATH without using `export`.

Reply:
It's better to save the path in .bashrc. So we should run:
`echo "export PATH=~/work/sof/xtensa-byt-elf/bin/:$PATH" >> ~/.bashrc`

Reply:
... except some users and/or distributions may prefer `.bash_profile` or something else (https://superuser.com/a/1452523), and then we risk this becoming yet another Unix guide. Can we discuss this in a future PR?

(the export is still not needed, PATH is already exported)